File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.c
void setZeroes(int** matrix, int matrixSize, int* matrixColSize) {
    int rows = matrixSize;
    int cols = matrixColSize[0];

    int zeroRow[rows];
    int zeroCol[cols];

    // Initially, no row or column contains zero
    for (int i = 0; i < rows; i++) {
        zeroRow[i] = 0;
    }

    for (int j = 0; j < cols; j++) {
        zeroCol[j] = 0;
    }

    // Find all rows and columns containing 0
    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < cols; j++) {

            if (matrix[i][j] == 0) {
                zeroRow[i] = 1;
                zeroCol[j] = 1;
            }
        }
    }

    // Set complete rows to 0
    for (int i = 0; i < rows; i++) {
        if (zeroRow[i] == 1) {
            for (int j = 0; j < cols; j++) {
                matrix[i][j] = 0;
            }
        }
    }

    // Set complete columns to 0
    for (int j = 0; j < cols; j++) {
        if (zeroCol[j] == 1) {
            for (int i = 0; i < rows; i++) {
                matrix[i][j] = 0;
            }
        }
    }
}

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.py
class Solution(object):

    def setZeroes(self, matrix):

        r = len(matrix)
        c = len(matrix[0])

        result = []

        # Find positions of original zeros
        for i in range(r):
            for j in range(c):
                if matrix[i][j] == 0:
                    result.append((i, j))

        # Make corresponding rows and columns zero
        for i, j in result:

            # Row
            for k in range(c):
                matrix[i][k] = 0

            # Column
            for l in range(r):
                matrix[l][j] = 0
        